bigbluebutton-html5/imports/api/meetings/server/modifiers/updateRandomViewer.js

```diff
@@ -2,27 +2,53 @@ import Meetings from '/imports/api/meetings';
 import Logger from '/imports/startup/server/logger';
 import { check } from 'meteor/check';
 
-export default function updateRandomUser(meetingId, userId, requesterId) {
+export default function updateRandomUser(meetingId, userIds, choice, requesterId) {
   check(meetingId, String);
-  check(userId, String);
+  check(userIds, Array);
+  check(choice, Number);
   check(requesterId, String);
 
   const selector = {
     meetingId,
   };
 
+  const userList = [];
+  if (choice < 0) { // no viewer
+    userList.push([requesterId,0]);
+  } else if (userIds.length == 1) {
+    userList.push([userIds[0],0]);
+  } else {
+    const intervals = [0, 200, 450, 750, 1100, 1500];
+    while (intervals.length > 0) {
+      const userId = userIds[Math.floor(Math.random() * userIds.length )];
+      if (userList.length != 0 && userList[userList.length-1][0] == userId) {// prevent same viewer from being selected sequentially
+        continue;
+      }
+      userList.push([userId, intervals.shift()]);
+    }
+    userList[userList.length-1][0] = userIds[choice]; // last one should be chosen in akka-app
+  }
+
+  if (userIds.length == 2) {
+    // I don't like this.. When the userList is same as previous one, we need to change it a bit to make sure that clients does not skip the change in MongoDB
```

Comment:
The current implementation chooses a candidate for 6 times randomly (with a restriction of an identical person not to be chosen sequentially) and makes a candidate list. Then the last one in the list is replaced with the one that was already chosen in akka-app. If the number of users is more than 2, the list would very rarely be the same. For example, the probability of the same list is 1/3 x 1/3 x 1/3 x 1/3 x 1/3 x 1/3 => 0.1% in case of 3 users. 
However, when there are only two users, the list can be the same with 25 % probability because only the first random choice and the akka-app's choice account for the list composition.
What I observed is that a Mongo DB update with the same list sometimes skips the re-rendering. In other words, only when there is two users the fancy selection "sometimes" does not start in the client's display, simply showing the final choice which came from the previous fancy selection. The displayed choice is correct anyway.
Therefore, I change the first item of the candidate list only when there are two users and the list turns completely the same with the previous one, so that the Mongo DB entry would be changed to force re-rendering. 
Probably this is not the best implementation, but would be a neat workaround without a large modification in other part of the code.